tests/agents/gherkin-specs/outcome.feature

```diff
@@ -1,55 +1,53 @@
 Feature: Outcome
 
+  Background: An agent with default configuration
+    Given an agent
+
   # ---- user set outcome
 
   Scenario: User set outcome on span has priority over instrumentation
-    Given an agent
-    And an active span
+    Given an active span
     And user sets span outcome to 'failure'
-    And span terminates with outcome 'success'
+    When span terminates with outcome 'success'
     Then span outcome is 'failure'
 
   Scenario: User set outcome on transaction has priority over instrumentation
-    Given an agent
-    And an active transaction
+    Given an active transaction
     And user sets transaction outcome to 'unknown'
-    And transaction terminates with outcome 'failure'
+    When transaction terminates with outcome 'failure'
     Then transaction outcome is 'unknown'
 
   # ---- span & transaction outcome from reported errors
 
   Scenario: span with error
-    Given an agent
-    And an active span
-    And span terminates with an error
+    Given an active span
+    When span terminates with an error
```

Comment:
[suggestion] We might even split `When span terminates with an error` in two parts:

```
Given an active span
And an error is reported to the span
When span terminates
```

That would allow to reuse the `When span terminates`. 

For `an error is reported to the span`, the "span" would be implicitly be the active span in `an error is reported to the span`

Reply:
nit: change `terminates` with `ends` as that's closer to the actual `span.end()` method.